src/unix/fs.c

```diff
@@ -88,10 +86,8 @@
       req->path = path;                                                       \
     } else {                                                                  \
       req->path = uv__strdup(path);                                           \
-      if (req->path == NULL) {                                                \
-        uv__req_unregister(loop, req);                                        \
+      if (req->path == NULL)                                                  \
         return UV_ENOMEM;                                                     \
```

Comment:
Now that I look at it... I think there is an accidental `req->bufs`  memory leak here.  We free it in `uv__fs_work()` but that doesn't run on error and `uv_fs_req_cleanup()` doesn't free it at all.

Suggestion: set `req->bufs = NULL` in INIT and add a `if (req->bufs != req->bufsml) uv__free(req->bufs)` stanza in `uv_fs_req_cleanup()`.

Existing bug so fine to do in a separate commit or PR.

Reply:
I'll send a follow up PR.